device_doctor/lib/src/config.dart

```diff
@@ -0,0 +1,40 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+
+import 'device.dart';
+
+Config _config;
+Config get config => _config;
+
+/// Configures device operating system based on injected decice OS.
+///
+/// This determines the [DeviceDiscovery].
+class Config {
+  Config({
+    @required this.deviceOperatingSystem,
+  });
+
+  static void initialize(String deviceOS) {
```

Comment:
Consider not using a global variable for passing around configuration. Its easy to track now when this project is small but will become difficult to refactor as it grows larger

Reply:
Updated to use local variable instead.